core/src/main/scala/org/apache/spark/util/collection/ExternalAppendOnlyMap.scala

```diff
@@ -463,21 +463,21 @@ class ExternalAppendOnlyMap[K, V, C](
 
     // An intermediate stream that reads from exactly one batch
     // This guards against pre-fetching and other arbitrary behavior of higher level streams
-    private var deserializeStream = nextBatchStream()
+    private var deserializeStream: Option[DeserializationStream] = None
```

Comment:
Perhaps `private var deserializeStream: Option[DeserializationStream] = nextBatchStream()` ? Then we can avoid many other code changes.

Reply:
Thanks @jiangxb1987 But this may make deserializeStream still init when spilling and generate DiskMapIterator

Reply:
can't we just use `null` as initial value? For performance critical places we don't need follow Scala style and use `None`.

Reply:
Ok,i will fix later @jiangxb1987 @cloud-fan Thanks for your precious time